[FSDP][2/N] Move `fsdp_modules(root_only=False)` -> `_get_fsdp_states()`

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #90959 [FSDP][6/N] Add note explaining idioms for `_FSDPState` traversal
* #90874 [FSDP][5/N] Add manual "wrapping" support for `fully_shard`
* #90871 [FSDP][4/N] Refactor func to share state/init handle attrs
* #90862 [FSDP][3/N] Move `fsdp_modules(root_only=True)` -> `_get_fsdp_root_states()`
* **#90861 [FSDP][2/N] Move `fsdp_modules(root_only=False)` -> `_get_fsdp_states()`**
* #90864 [FSDP][Easy] Rename `entry` -> `fsdp_module` to be more descriptive
* #90860 [FSDP][1/N] Add `_get_fsdp_states()`

This PR migrates all internal usages of `FullyShardedDataParallel.fsdp_modules(root_only=False)` to `_get_fsdp_states()`. This is to unify the code paths for composable and wrapper FSDP.

This PR _does not_ change the usages in test files. This is because we should revisit those usages separately as a way to track which functionality for which we have not tested composable FSDP.

Commit:
[FSDP][2/N] Move `fsdp_modules(root_only=False)` -> `_get_fsdp_states()`

[ghstack-poisoned]

Commit:
Update on "[FSDP][2/N] Move `fsdp_modules(root_only=False)` -> `_get_fsdp_states()`"


This PR migrates all internal usages of `FullyShardedDataParallel.fsdp_modules(root_only=False)` to `_get_fsdp_states()`. This is to unify the code paths for composable and wrapper FSDP.

This PR _does not_ change the usages in test files. This is because we should revisit those usages separately as a way to track which functionality for which we have not tested composable FSDP.

[ghstack-poisoned]

Commit:
Update on "[FSDP][2/N] Move `fsdp_modules(root_only=False)` -> `_get_fsdp_states()`"


This PR migrates all internal usages of `FullyShardedDataParallel.fsdp_modules(root_only=False)` to `_get_fsdp_states()`. This is to unify the code paths for composable and wrapper FSDP.

This PR _does not_ change the usages in test files. This is because we should revisit those usages separately as a way to track which functionality for which we have not tested composable FSDP.

[ghstack-poisoned]

rohan-varma reviewed (approved):
The change itself looks good, but curious how we will enforce this for future developers? I'm wondering if we can add some unittesting that could catch the class-based programing automatically instead of relying on reviewers which is prone to human error.

Commit:
Update on "[FSDP][2/N] Move `fsdp_modules(root_only=False)` -> `_get_fsdp_states()`"


This PR migrates all internal usages of `FullyShardedDataParallel.fsdp_modules(root_only=False)` to `_get_fsdp_states()`. This is to unify the code paths for composable and wrapper FSDP.

This PR _does not_ change the usages in test files. This is because we should revisit those usages separately as a way to track which functionality for which we have not tested composable FSDP.

[ghstack-poisoned]

Commit:
Update on "[FSDP][2/N] Move `fsdp_modules(root_only=False)` -> `_get_fsdp_states()`"


This PR migrates all internal usages of `FullyShardedDataParallel.fsdp_modules(root_only=False)` to `_get_fsdp_states()`. This is to unify the code paths for composable and wrapper FSDP.

This PR _does not_ change the usages in test files. This is because we should revisit those usages separately as a way to track which functionality for which we have not tested composable FSDP.

[ghstack-poisoned]

Commit:
Update on "[FSDP][2/N] Move `fsdp_modules(root_only=False)` -> `_get_fsdp_states()`"


This PR migrates all internal usages of `FullyShardedDataParallel.fsdp_modules(root_only=False)` to `_get_fsdp_states()`. This is to unify the code paths for composable and wrapper FSDP.

This PR _does not_ change the usages in test files. This is because we should revisit those usages separately as a way to track which functionality for which we have not tested composable FSDP.

[ghstack-poisoned]

Commit:
Update on "[FSDP][2/N] Move `fsdp_modules(root_only=False)` -> `_get_fsdp_states()`"


This PR migrates all internal usages of `FullyShardedDataParallel.fsdp_modules(root_only=False)` to `_get_fsdp_states()`. This is to unify the code paths for composable and wrapper FSDP.

This PR _does not_ change the usages in test files. This is because we should revisit those usages separately as a way to track which functionality for which we have not tested composable FSDP.

[ghstack-poisoned]

awgu commented:
@pytorchbot merge
